### PR TITLE
fix(libsinsp_e2e): make forking_clone_fs more reliable

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

> /area libsinsp

/area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
> [  FAILED  ] sys_call_test.forking_clone_fs

:skull:

File: test/libsinsp_e2e/forking.cpp [hunk 1]
```diff
@@ -336,7 +336,7 @@ static int clone_callback_1(void* arg)
 	// using a weird clone() here something goes wrong with
 	// recent version of glibc
 	ctid = syscall(SYS_getpid);
-	close(cp->fd);
+	printf("CHILD %d\n", close(cp->fd));
```
Comment (LucaGuerra) [inline test/libsinsp_e2e/forking.cpp:339]: leftover or should we display it?

File: test/libsinsp_e2e/forking.cpp [hunk 2]
```diff
@@ -392,19 +395,20 @@ TEST_F(sys_call_test, forking_clone_fs)
...
 		/* Parent falls through to here. Wait for child; __WCLONE option is
 		   required for child notifying with signal other than SIGCHLD. */
 
-		pid = waitpid(-1, &status, __WCLONE);
+		pid = waitpid(clone_tid, &status, __WCLONE);
 		if (pid == -1)
 			FAIL();
 
-		close(cp.fd);
+		printf("PARENT %d\n", close(cp.fd));
```
Comment (LucaGuerra) [inline test/libsinsp_e2e/forking.cpp:410]: same as above

File: test/libsinsp_e2e/forking.cpp (hunk 1 above)
Comment (therealbobo) [reply test/libsinsp_e2e/forking.cpp:339]: Sadly this does not fix the bug :/ I can't get a repro on my local setup and I'm trying to get some more info with the CI 😅 I'll remove this debug prints obv :)

Comment (LucaGuerra) [reply test/libsinsp_e2e/forking.cpp:339]: Thanks @therealbobo ! Could you please add `wip: ` to the PR title until you are done with experiments?

Comment (therealbobo) [reply test/libsinsp_e2e/forking.cpp:339]: sure!

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**